src/esp/physics/PhysicsManager.cpp

```diff
@@ -84,10 +84,7 @@ int PhysicsManager::addObject(const std::string& configFile,
 }
 
 int PhysicsManager::removeObject(const int physObjectID) {
-  if (existingObjects_.count(physObjectID) == 0) {
-    LOG(ERROR) << "Failed to remove object: no object with ID " << physObjectID;
-    return ID_UNDEFINED;
-  }
+  assertIDValidity(physObjectID);
```

Comment:
Should the object destructor call removeObject()? That way you never forget to call it.
delete should be avoid and RTTI used instead. Could we use unique_ptr instead here? The last three lines would then boil down to:

existingObjects_.erase(physObjectID);

Reply:
This comes back to the SceneNode problem. SceneNode objects are owned completely by the SceneGraph. I have not gotten around to converting RigidObject from a SceneNode to an AbstractFeature3D as suggested in reviews for PR #91 . We use raw pointers to denote that PhysicsManager does not own the RigidObject. 

That said, the removal process could probably be improved. I'll lump that in with the AbstractFeature3D re-design.

Reply:
If we don't own the object, why are we calling delete on it below on line 89?

Reply:
We have to remove the SceneNode from the SceneGraph when the object no longer exists. I agree the current situation is not entirely self-consistent. The simulator was not initially designed for this sort of dynamic restructuring.